### PR TITLE
Add opencost ui to deployment

### PR DESCRIPTION
**Pull request by jwcesign**
Signed-off-by: jiang wei <jwcesign@gmail.com>

## What does this PR change?
* Now deploy the `ui` in opencost is not clearly, we need this for trying
* I don't know how to push the img from my repo to quay.io, maybe anyone can help about this? it's build based on `ui` and nginx

## Does this PR relate to any other PRs?
* none

## How will this PR impact users?
* none

## Does this PR address any GitHub or Zendesk issues?
* none

## How was this PR tested?
* deploy

## Does this PR require changes to documentation?
* Yes, will change latter

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?
* none

### CODE REVIEW

**Comment by jwcesign** [comment]
cc @nikovacevic

**Comment by dwbrown2** [comment]
@jwcesign can you share any more context on the goals around this PR? wasn't able to follow based on this description. 

```
Now deploy the ui in opencost is not clearly, we need this for trying
```

thanks for your contribution!

**Comment by jwcesign** [comment]
1. If want to deploy the `ui`, we need to install npm on host machine, but it's better to deploy with container.
2. it's better to deploy `ui` with one `kubectl` command and also, it's better to install it default(So I write it in `opencost.yaml`).

File: kubernetes/opencost.yaml [hunk 1]
```diff
@@ -139,7 +166,27 @@ spec:
...
       serviceAccountName: opencost
+      volumes:
+        - name: nginx-conf
+          configMap:
+            name: nginx-conf
+            items:
+              - key: nginx.conf
+                path: default
       containers:
+        - image: cesign/opencost-ui:v0.12
```
Comment (jessegoodier) [inline kubernetes/opencost.yaml]: Do you have the dockerfile that goes with this image? Maybe add to the /ui folder?

Comment (jwcesign) [reply kubernetes/opencost.yaml]: fixed

**Comment by jwcesign** [comment]
So after this merged, we should build a image(in dir `opencost/ui`) based on the docker file:
```shell
docker build -t quay.io/kubecost1/opencost-ui:latest -f Dockerfile ./
```
And then push it

**Comment by jwcesign** [comment]
cc @jessegoodier

**Comment by biancaburtoiu** [comment]
Looking into this to gather more context before approving and merging. Will update ASAP

**Comment by biancaburtoiu** [comment]
I have discussed this with one other engineer (cc @nealormsbee) and we'll have to make several adjustments on our side, beyond this PR, to offer a UI container as part of the opencost deployment. We'll include the same changes submitted here but not this PR as-is. 

I will update on this same thread once we've got the updated opencost deployment ready to go. Thank you again for your contribution @jwcesign!

**Comment by dudicoco** [comment]
Any updates on this? The docs are currently very misleading as the main README has a pic of the UI, however that UI is not available within the cluster which is highly desirable by everyone.

**Comment by nealormsbee** [comment]
Hi @dudicoco 👋 

There has been movement here; my apologies to the community slow communication. When @biancaburtoiu and I spoke about this contribution, I raised a few issues. Namely:

- This PR would run the `parcel` development server behind NGINX. Better practice would be to build the UI bundle and have NGINX serve that content statically.
- Rearranging the project to nest a bunch of things under a `ui/server` subdirectory seemed unnecessary.
- The addition to `kubernetes/opencost.yaml` references an `opencost-ui` image, which doesn't exist yet. It can, but we'd need to update our build automation to create and host an `opencost-ui` image.

Since then, it looks like we've added a `ui/Dockerfile` that addresses the first and second concerns.

I agree that it probably makes sense for `opencost` to supply yaml to deploy this image. Exactly how we do that (always deploy the ui image? Provide two yaml files, one that deploys full stack and one that deploys just the metrics agent?) I don't have much of an opinion on. @jessegoodier @michaelmdresser any thoughts here?


Overall, I'd vote to:

- Close this PR.
- Open an issue for adding some yaml as an easy option to deploy the opencost UI.
- Update our build processes to host an opencost UI image, which would be updated monthly along with our other images.

**Comment by michaelmdresser** [comment]
My guess is that @mattray should be aware of this, he may have more up-to-date thoughts than I do!

**Comment by mattray** [comment]
I agree that we should get the UI image published and added to the yaml for deployment. Let's close this issue in favor of https://github.com/opencost/opencost/pull/1411.

**Pull request closed without merging**